### ISSUE TITLE
Reflectable transformer fails when there is an enum present

### ISSUE THREAD

**Issue by luisvt** [dart-lang/reflectable#80]
when there is an enum in the entry_point file transformer fails and throws an error similar to this one:

```
Build error:
Transform Reflectable on dson|test/serializer/serialize_enums_test.dart threw error: Class 'CompilationUnitImpl' has no instance getter 'metadata'.

NoSuchMethodError: method not found: 'metadata'
Receiver: Instance of 'CompilationUnitImpl'
Arguments: []
dart:core-patch/object_patch.dart 42                          Object._noSuchMethod
dart:core-patch/object_patch.dart 45                          Object.noSuchMethod
package:reflectable/src/transformer_implementation.dart 4216  _extractMetadataCode
package:reflectable/src/transformer_implementation.dart 1570  _ReflectorDomain._fieldMirrorCode
package:reflectable/src/transformer_implementation.dart 987   _ReflectorDomain._generateCode.<fn>
```

the code that I'm using is next one:
```
import 'package:dson/dson.dart';
import 'package:test/test.dart';

enum Color {
  BLUE,
  RED,
  GREEN
}

@serializable
class ObjectWithEnum {
  Color color;
}

main() {

  test('deserialize enum', () {
    expect(fromJson('0', Color), Color.BLUE);
  });

  test('deserialize object with enum', () {
    expect(fromJson('{"color":0}', ObjectWithEnum).color, Color.BLUE);
    expect(fromJson('{"color":1}', ObjectWithEnum).color, Color.RED);
    expect(fromJson('{"color":2}', ObjectWithEnum).color, Color.GREEN);
  });
}
```

and serializable declaration is next one:

```
/// This annotation specifies the object can be serialized
class Serializable extends Reflectable {
  const Serializable() : super(
      typeAnnotationQuantifyCapability,
      superclassQuantifyCapability,
      invokingCapability,
      declarationsCapability,
      metadataCapability,
      newInstanceCapability,
      typeRelationsCapability,
      typeCapability);
}

/// This annotation specifies the object can be serialized
const serializable = const Serializable();
```

**Comment by eernstg** [comment]
Fixed by 79002470672b576842afda4b0a9f899aa1d3bf69.

**Comment by eernstg** [comment]
Thanks for the input, this fix will be part of an upcoming 0.5.4.
